kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -0,0 +1,575 @@
+    }
+    
+    public void givenNetworkConfigMapWith(String key, Object value) {
+        netConfig.put(key, value);
+    }
+    
+    public void givenNetworkServiceMockedForUsbInterfaceWithName(String interfaceName) {
+        this.networkService = mock(NetworkService.class);
+        
+        UsbNetDevice mockUsbNetDevice = mock(UsbNetDevice.class, RETURNS_DEEP_STUBS);
+        
+        when(this.networkService.getUsbNetDevice(interfaceName)).thenReturn(Optional.of(mockUsbNetDevice));
+    }
+    
+    public void givenNetworkServiceThatAlwaysReturnsEmpty() {
+        this.networkService = mock(NetworkService.class);
+        when(this.networkService.getUsbNetDevice(any())).thenReturn(Optional.empty());
+    }
+
+    public void whenGetDbusConnectionIsRun() {
+        this.dbusConnectionInternal = this.instanceNMDbusConnector.getDbusConnection();
+    }
+
+    public void whenCheckPermissionsIsRun() {
+        this.instanceNMDbusConnector.checkPermissions();
+    }
+
+    public void whenCheckVersionIsRun() {
+        try {
+            this.instanceNMDbusConnector.checkVersion();
+        } catch (DBusException e) {
+            hasDBusExceptionBeenThrown = true;
+        }
+    }
+
+    public void whenGetInterfaces() {
+        try {
+            this.internalStringList = this.instanceNMDbusConnector.getInterfaces();
+        } catch (DBusException e) {
+            hasDBusExceptionBeenThrown = true;
+        }
+    }
+
+    public void whenApply() {
+        try {
+            this.instanceNMDbusConnector.apply();
+        } catch (DBusException e) {
+            hasDBusExceptionBeenThrown = true;
+        }
+    }
+
+    public void whenApplyIsCalledWith(Map<String, Object> networkConfig) {
+        try {
+            this.instanceNMDbusConnector.apply(networkConfig);
+        } catch (DBusException e) {
+            e.printStackTrace();
+            hasDBusExceptionBeenThrown = true;
+        } catch (NoSuchElementException e) {
+            e.printStackTrace();
+            hasNoSuchElementExceptionThrown = true;
+        } catch (NullPointerException e) {
+            e.printStackTrace();
+            hasNullPointerExceptionThrown = true;
+        }
+    }
+    
+    public void whenGetInterfaceStatus(String netInterface, NetworkService netService) {
+        try {
+            this.netInterface = this.instanceNMDbusConnector.getInterfaceStatus(netInterface, netService);
+        } catch (DBusException e) {
+            e.printStackTrace();
+            hasDBusExceptionBeenThrown = true;
+        } catch (NoSuchElementException e) {
+            e.printStackTrace();
+            hasNoSuchElementExceptionThrown = true;
+        } catch (NullPointerException e) {
+            e.printStackTrace();
+            hasNullPointerExceptionThrown = true;
+        }
+    }
+
+    public void thenNoExceptionIsThrown() {
+        assertFalse(hasDBusExceptionBeenThrown);
+        assertFalse(hasNoSuchElementExceptionThrown);
```

Comment:
You forgot about the NPE exception:

```suggestion
        assertFalse(hasNoSuchElementExceptionThrown);
        assertFalse(hasNullPointerExceptionThrown);
```